google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java

```
@@ -685,6 +697,25 @@ private void fillField(
         }
         break;
       case STRING:
+        // BQ Table Timestamp Fields are transmitted as a String. Check this first
+        // as numeric values may actually represent a timestamp (micros from epoch).
+        // Supported types: https://docs.cloud.google.com/bigquery/docs/supported-data-types
+        if (fieldSchema != null && fieldSchema.getType() == TableFieldSchema.Type.TIMESTAMP) {
+          if (val instanceof String) {
+            String value = (String) val;
+            if (isValidTimestamp(value)) {
```

Could we throw an exception if this is false?